src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java

```diff
@@ -154,7 +155,7 @@ public KendraIntelligentRankingProperties(final List<String> bodyFields,
 
     public KendraIntelligentRankingProperties(StreamInput input) throws IOException {
       this.bodyFields = input.readStringList();
-      this.bodyFields = input.readStringList();
+      this.titleFields = input.readStringList();
```

Comment:
is this related to the XContent classes change?

Reply:
Not directly. 

When merging the cherry-picked commit, I also picked up the unit tests that I had previously added (in KendraIntelligentRankingConfigurationTests).

When I added those tests in `main`, they also revealed some bugs (like this one). I should go through and make sure that all of the tests I added in `main` get backported to 2.x, since I fixed a bunch of little bugs along the way.

Reply:
Oh... I just added the `backport 2.x` label to https://github.com/opensearch-project/search-processor/pull/99.

Reply:
Oh, but the backport PR fails to build because we need to merge this PR to fix the XContent stuff:

https://github.com/opensearch-project/search-processor/pull/126

Reply:
> Oh, but the backport PR fails to build because we need to merge this PR to fix the XContent stuff:
> 
> #126

Looks like there are two ways to handle this issue:
1. In this PR, remove the content of #99, merge this one and merge https://github.com/opensearch-project/search-processor/pull/126
2. Change the description to include the backport of #99, then merge this one; the backport PR https://github.com/opensearch-project/search-processor/pull/126 is pointless in this case because the code is already there.

I'm fine with either, but `1.` appears to overreact, and the order of commits in 2.x is messed up comparing to main branch

Reply:
> Oh, but the backport PR fails to build because we need to merge this PR to fix the XContent stuff:
> 
> #126

Looks like there are two ways to handle this issue:
1. In this PR, remove the content of #99, merge this one and merge https://github.com/opensearch-project/search-processor/pull/126
2. Change the description to include the backport of #99, then merge this one; the backport PR https://github.com/opensearch-project/search-processor/pull/126 is pointless in this case because the code is already there.

I'm fine with either, but `1.` appears to overreact, and the order of commits in 2.x is messed up comparing to main branch

Reply:
I think #126 shouldn't have any conflicts once this one is merged, because any overlapping changes should be identical.